File: backend/src/main/java/com/smartpark/service/PassService.java
package com.smartpark.service;

import com.smartpark.dto.request.ParkingPassRequest;
import com.smartpark.dto.response.ParkingPassResponse;

import java.util.List;

public interface PassService {
    ParkingPassResponse purchasePass(Long userId, ParkingPassRequest request);
    List<ParkingPassResponse> getUserPasses(Long userId);
    ParkingPassResponse getPassById(Long passId, Long userId);
    List<ParkingPassResponse> getAllPasses();
    void cancelPass(Long passId, Long userId);
}

File: frontend/src/services/passService.js
import api from './api';

export const passService = {
  purchasePass: async (data) => {
    return api.post('/passes', data);
  },

  getMyPasses: async () => {
    return api.get('/passes/my');
  },

  getPassById: async (id) => {
    return api.get(`/passes/${id}`);
  },

  cancelPass: async (id) => {
    return api.put(`/passes/${id}/cancel`);
  }
};
